React assignment 4

abhishekpatel946 opened the pull request:
### React Form Assignment

- [x] Components further divided into small components. 
- [x] SCSS use for each component styling.
- [x] Get the form data onSubmit. 
- [x] Validate the form data dynamically.
- [x] check the proptype for each component.
- [x] data store in redux after validation.
- [x] for backup the data stored in the localStorage from the redux state.
- [x] data displayed from the redux state directly.

abhishekpatel946 committed:
Initialize project using Create React App

abhishekpatel946 committed:
initial commit

Commit to push code on GitHub

abhishekpatel946 committed:
fix: Conflict Resolved from ReadME

abhishekpatel946 committed:
feat: add SCSS on App

SCSS main-file and colorStack for whole App to reduce file from stucture

abhishekpatel946 committed:
fix: Form with validation

form grab the data from the user and checks the basic typos error in btw filling the form

abhishekpatel946 committed:
fix: SCSS for individual module

break the main.scss for individual components and change the property using it, hence the code for SCSS is much smaller for each component and readable

abhishekpatel946 committed:
feat: add PasswordStrengthBar

check for strenght of the password when user type in the inputbox for password

abhishekpatel946 committed:
feat: create ReadME.md file

ReadME file for demo and deployed link is available in it, and short summary of the problem and with solution code

hrathi22 commented on src/Components/Form/InputText.jsx:
```diff
@@ -0,0 +1,30 @@
+
+const InputText = (props) => {
+  // typechecking with propTypes
+  InputText.propTypes = {
```
An incorrect place to add prop validation check https://reactjs.org/docs/typechecking-with-proptypes.html

hrathi22 replied:
also add prop type for all variables

hrathi22 commented on src/Components/Form/InputText.jsx:
```diff
@@ -0,0 +1,30 @@
+  const { type, value, name, placeholder, onChange } = props;
+
+  return (
+    <div className='InputText'>
```
it can be written as `inputText` and make changes for all css/scss class name

hrathi22 commented on src/Components/Form/Label.jsx:
```diff
@@ -0,0 +1,18 @@
+  };
+
+  return (
+    <div>
```
DO we need div here?

hrathi22 commented on src/Components/Form/RadioButton.jsx:
```diff
@@ -0,0 +1,44 @@
+
+  return (
+    <div className='Radio-button'>
+      <input
```
Use json object for radio btn and loop it here, pass the json from HOC

hrathi22 commented on src/Components/Form/SelectOption.jsx:
```diff
@@ -0,0 +1,32 @@
+        className='Input-name'
+        name={name}
+        value={value}
+        onChange={onChange}>
```
Use json for options and pass json from HOC

hrathi22 commented on src/Components/Container/FormContainer.jsx:
```diff
@@ -0,0 +1,159 @@
+  const handleSubmit = (e) => {
+    // validate the form
+    if (!name) {
+      <Error msg={'error prompt for Name'} />;
```
why We are using Error component here?

hrathi22 commented on src/Components/Container/FormContainer.jsx:
```diff
@@ -0,0 +1,159 @@
+
+  // change handler for states
+  const changeHandlerName = (e) => {
+    setName(e.target.value);
```
Use forms on submit an event to get values instead of using state for each form element

hrathi22 commented on src/Components/Container/FormContainer.jsx:
```diff
@@ -0,0 +1,159 @@
+    } else if (confirmPassword !== password) {
+      alert("Password's are not matched!");
+    } else {
+      alert('Form Validation Successful.');
```
Use const's (create separate file) for all magic strings and numbers

hrathi22 commented on src/Components/ErrorPrompt/Error.jsx:
```diff
@@ -0,0 +1,18 @@
+
+  console.log(props);
+
+  return <div>{alert({ msg })}</div>;
```
Not a proper way to alert for any error message

hrathi22 commented on src/Components/Form/Button.jsx:
```diff
@@ -0,0 +1,28 @@
+      <button
+        className={className}
+        type={type}
+        // onClick={onClick}
```
make button component reusable, by using on submit we will be able to use this component for forms only

abhishekpatel946 committed:
fix: change after code review

css/scss property name as small, unused code be removed, error component removed, form is onSubmit now instead of onClick, make button component reusable, change the radio & select component become more generic to use

abhishekpatel946 committed:
fix: demo img in ReadMe

link the correct demp images from the github/repo/assets

abhishekpatel946 committed:
feat: add localStorage

to store the formData into localStorage for persist data

abhishekpatel946 committed:
fix: DatePicker

materialUI datepicker used and handle the event.target.value if undefined error occurrs

abhishekpatel946 committed:
feat: materialUI table

table is responsive and handle the data with some sort of SCSS

abhishekpatel946 committed:
fix: add localStorage

getter & setter method is in the DataService for localStorage and pass the data using props to the table component for display stored data, and new data stored into the localStorage after the validation check

abhishekpatel946 committed:
fix: reduce the validation code

abhishekpatel946 committed:
fix: dataService for get/set request

dataService is working with localStorage

abhishekpatel946 committed:
fix: fixing a bug into dataService

abhishekpatel946 committed:
fix: app handle the undefined or null value in localStorage

abhishekpatel946 committed:
fix: Update Code after Review

abhishekpatel946 committed:
fix: dataService & its operations

rewrite the dataService and add the deleteItem and updateItem functionality

abhishekpatel946 committed:
fix: dataService & deleteItem function

stable version of CRUD-ops with form, editItem & mapping on data is not working in this version

abhishekpatel946 committed:
feat: add new project

instead of old code add the new code is working fine

abhishekpatel946 committed:
docs: Bug Report updates

abhishekpatel946 committed:
feat: change file structure & exporting

Update the file structure more consice and added index for export each and every component in one short hand, also debug the code for initialState is creating a problem for us.

abhishekpatel946 committed:
fix: updation issue

abhishekpatel946 committed:
issue with localStorage to set & get the data, and also re-rendering problem.

abhishekpatel946 committed:
feat: done with the assignment

Form with localStorage to store the data with validation check and password strength is included.

abhishekpatel946 committed:
Update README.md

add another pending task and deployed link

abhishekpatel946 committed:
feat: add React-Redux for state management

abhishekpatel946 committed:
add change in ReadMe

abhishekpatel946 committed:
feat: redux with create & delete using state

with the help of redux add & delete user data using states, no localStorage used here

abhishekpatel946 committed:
fix: update state using redux

abhishekpatel946 committed:
docs: add file structure in ReadME

abhishekpatel946 committed:
fix: dataService for setItem

addUser, deleteUser & updateUser all service working fine, but execute one step later

abhishekpatel946 committed:
feat: CRUD Redux with localStorage

addUser, deleteUser & updateUser work perfectly using Redux and data backup with localStorage

abhishekpatel946 committed:
docs: edit pending task in ReadME

The pending task is completed for Training_Day_8-9.

abhishekpatel946 commented:
**Complete React_Training_Day_8-9_Integration with localStorage and perform CRUD operations.**
